lib/shared/binary-file-utils.ts

```diff
@@ -0,0 +1,51 @@
+import * as path from "node:path"
```

Comment:
never create "utils" files- ALWAYS name files after the main function export e.g. `isBinaryFile` and move `hasBinaryContent` to a separate file